source/common/http/http2/codec_impl.cc

```diff
@@ -357,6 +358,23 @@ void ConnectionImpl::StreamImpl::encodeMetadata(const MetadataMapVector& metadat
 void ConnectionImpl::StreamImpl::processBufferedData() {
   ENVOY_CONN_LOG(debug, "Stream {} processing buffered data.", parent_.connection_, stream_id_);
 
+  // Restore crash dump context when processing buffered data.
+  Event::Dispatcher& dispatcher = parent_.connection_.dispatcher();
+  ASSERT(dispatcher.trackedObjectStackIsEmpty());
+  Envoy::ScopeTrackedObjectStack stack;
+  stack.add(parent_.connection_);
+
+  absl::Cleanup clear_current_stream_id = [this]() { parent_.current_stream_id_.reset(); };
+  // TODO(kbaichoo): When we add support to *ConnectionImpl::getStream* for
+  // deferred closed streams we can use their stream id here.
+  if (!stream_manager_.buffered_on_stream_close_) {
+    ASSERT(!parent_.current_stream_id_.has_value());
+    parent_.current_stream_id_.emplace(stream_id_);
```

Comment:
out of curiosity, is emplace better than simple = here (since stream_id is a uint32_t I would think = should be just as performant?)

Reply:
Up to you, but I'd be inclined to use = unless emplace() is better for some reason? Your call though.

Reply:
done.